Rename tensor mechanics module and use Physics syntax

GiudGiud opened the pull request:
refs #25584
A few things to decide with the reviewers.

- new name for CohesiveZoneMaster?
- new name for LineElementMaster?
- Do we remove the Kernels/TensorMechanicsMaster and Kernels/DynamicTensorMechanics syntax now? It could simplify this PR a bit leaving only 2 valid syntaxes

To do list:
- [x] Make patches
- [x] Review my own PR
- [x] Finalize the Physics names
- [x] Get reviews
- [x] Merge testing recipe change
- [x] Address failures in recipes enabled
- [x] Get this merged
- [x] Get all patches merged

App patches:
- [x] BlueCRAB https://github.inl.gov/ncrc/blue_crab/pull/246
- [x] Direwolf https://github.inl.gov/ncrc/dire_wolf/pull/289
- [x] Griffin https://github.inl.gov/ncrc/griffin/pull/1759
- [x] SAM https://git-nse.egs.anl.gov/SAM/SAM/-/merge_requests/889
- [x] Marmot https://github.inl.gov/ncrc/marmot/pull/923
- [x] BlackBear https://github.com/idaholab/blackbear/pull/387
- [x] isopod https://github.com/idaholab/isopod/pull/90
- [x] malamute https://github.com/idaholab/malamute/pull/120
- [x] mastodon https://github.com/idaholab/mastodon/pull/447
- [x] tmap8 https://github.com/idaholab/TMAP8/pull/94
- [x] magpie https://github.com/idaholab/magpie/pull/496
- [x] Okami https://hpcgitlab.hpc.inl.gov/idaholab/okami/-/merge_requests/40
- [x] Redback https://github.com/pou036/redback/pull/198
- [ ] Ferret https://github.com/mangerij/ferret/pull/330
- [x] Racoon https://github.com/hugary1995/raccoon/pull/160
- [x] Grizzly https://github.inl.gov/ncrc/grizzly/pull/758
- [x] Bison https://github.inl.gov/ncrc/bison/pull/5833
- [x] Falcon https://github.com/idaholab/falcon/pull/142

GiudGiud committed:
Rename tensor mechanics module to solid mechanics, refs #25584

GiudGiud committed:
Update modules central makefile for new module name, refs #25584

GiudGiud committed:
Add symlinks for tensor mechanics module
This keeps additional civet recipes green during the transition

GiudGiud committed:
Rename doco folders to solid mechanics

GiudGiud committed:
Rename SQA and bib files to solid mechanics

GiudGiud committed:
Adapt SQA documents to new name

GiudGiud committed:
Adapt unit tests to new module name

moosebuild committed:
Register classes to new solid mechanics app

moosebuild committed:
Rename tensor mechanics UOs to solid mecha UOs

GiudGiud committed:
Adapt materials to new UO names

GiudGiud committed:
Adapt the app to new name

GiudGiud committed:
Adapt utils to new name

GiudGiud committed:
Adapt actions to new name

GiudGiud committed:
Update submodules

moosebuild committed:
Update test objects to new module name

GiudGiud committed:
Adapt other modules to new solid mechanics module

GiudGiud committed:
Add macro for deprecating syntax

GiudGiud committed:
Update action syntax to use Physics

GiudGiud committed:
Use Physics name for base classes of major SM actions

moosebuild committed:
Update tensor mechanics syntax in tests

GiudGiud committed:
Use QuasiStatic in physics name

moosebuild committed:
Update action syntax in examples
Update UO names in examples

moosebuild committed:
Update solid mechanics action syntax in examples

moosebuild committed:
Update solid mechanics UO names in examples

moosebuild committed:
Update solid mechanics UO names in tests

GiudGiud committed:
Update module path in SQA docs and in doc configuration

GiudGiud committed:
Update solid mechanics module name in MOOSE tutorial

GiudGiud committed:
Update doco source doco & links for new physics names

GiudGiud committed:
Update modules/solid_mech to new syntax

GiudGiud committed:
Update syntax doco to new syntax

GiudGiud committed:
Add back old pages for compatibility

GiudGiud committed:
Add new Physics syntax pages

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2044920/) on ce70dec wanted to post the following:

The following commit(s) contain fixups:

- 92cb8b1ec6: fixup! Adapt actions to new name

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2044973/) on 8defb67 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/26739/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/26739/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 1fbb5aad81de244e073f9e5ae406701ac6831edc`

GiudGiud committed:
Unrelated Clang format

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2046350/) on 0fdfc75 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/26739/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/26739/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 19aa8bed5a50249b7e3a2f499a45c192e4d8ddf9`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2046527/) on 4b01e2c wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/26739/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/26739/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 19aa8bed5a50249b7e3a2f499a45c192e4d8ddf9`

GiudGiud committed:
Update framework documentation to new module name

GiudGiud committed:
Update modules documentation to new modules names

moosebuild committed:
Replace TM by SolidMechanics in XFEM, Combined, Porous Flow and Contact

GiudGiud committed:
Fixup deprecated syntax doco to point to new tests names and blocks

GiudGiud committed:
Update documentation module items to new module name
- sqa reports
- duplicate bibs
- tutorial links
- gallery

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2051231/) on 93b691a wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/26739/site)

This comment will be updated on new commits.

GiudGiud committed:
Rename UO doc files

moosebuild committed:
(Dynamic)SolidMechanicsAction -> Dynamic/QuasiStaticSolidMechanicsPhysics

GiudGiud committed:
More backing up on bad replacements

GiudGiud committed:
Update tests spec design document location

moosebuild committed:
Add more pages for deprecated kernels pages

GiudGiud committed:
Attempt deleting doco symlink
- if we dont do that the 'child' class searching extension cannot find the children classes headers and errors

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2053215/) on 1e31014 wanted to post the following:

The following commit(s) contain fixups:

- 40caf82e94: fixup! More backing up on bad replacements
- fba7d17c49: fixup! Add back old pages for compatibility

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit fd4e77d5249ab0cf5614ef03ec589f16a199124d
+Subproject commit 199a23c71d556343cbac89ceffbc987428aa4464
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

GiudGiud committed:
Add two deprecated headers to keep the actions around until the apps are patched

GiudGiud committed:
Add doco for new name of parameter sharing action

GiudGiud committed:
Update module loader to new tensor mechanics module name
Makefile cosmetic changes

dschwen commented:
Please set

```
Author: MOOSE maintenance <mooseframework@inl.gov>
```

for the large rename commits.

GiudGiud commented:
Done! Should have caught all the big ones.

GiudGiud commented on modules/solid_mechanics/doc/content/syntax/Kernels/SolidMechanics/index.md:
```diff
@@ -0,0 +1,11 @@
+See [this page](/Kernels/TensorMechanics/index.md)
+
+!alert warning title=Deprecated Action
+This legacy action will soon be deprecated in favor of the more inclusive
```
done

bwspenc replied:
I don't see this changed in the update. If you want to keep it how it is, I would at least suggest changing it to say something like "more inclusively named" to unambiguously convey that point.

GiudGiud replied:
this wording wasnt fine. I've said new, which is true. If you prefer comprehensive I can change again
my bad for not pushing the change. I ended up looking at something else. it's online now

GiudGiud commented on modules/solid_mechanics/doc/content/syntax/Kernels/TensorMechanics/index.md:
```diff
@@ -1,55 +1,50 @@
+!syntax description /Kernels/SolidMechanics/LegacyTensorMechanicsAction
 
 !alert warning title=Deprecated Action
 This legacy action will soon be deprecated in favor of the more inclusive
```
done

bwspenc replied:
I don't see this changed either.

GiudGiud commented on modules/fsi/doc/content/modules/fsi/sqa/fsi_sdd.md:
```diff
@@ -14,7 +14,7 @@ Specific details unique to the module are outlined in this document.
 [software dependencies and limitations of the MOOSE framework](framework_sdd.md#dependencies-and-limitations),
 as well as the dependencies and limitations of the [Navier-Stokes](navier_stokes_sdd.md#dependencies-and-limitations)
-and [tensor mechanics](tensor_mechanics_sdd.md#dependencies-and-limitations) modules. The {{module}}
+and [solid mechanics](solid_mechanics_sdd.md#dependencies-and-limitations) modules. The {{module}}
```
these documents are standardized. if the module changes name we cannot refer to the old name

GiudGiud commented on modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_sdd.md:
```diff
@@ -14,7 +14,7 @@ Specific details unique to the module are outlined in this document.
 [software dependencies and limitations of the MOOSE framework](framework_sdd.md#dependencies-and-limitations),
 as well as the dependencies and limitations of the [Heat Transfer](heat_transfer_sdd.md#dependencies-and-limitations) module
-when performing coupled heat transfer simulations, the [Fluid Properties](tensor_mechanics_sdd.md#dependencies-and-limitations optional=True) module
+when performing coupled heat transfer simulations, the [Fluid Properties](solid_mechanics_sdd.md#dependencies-and-limitations optional=True) module
```
good catch!

bwspenc replied:
This didn't get fixed.

GiudGiud commented on modules/solid_mechanics/doc/content/syntax/Physics/SolidMechanics/GeneralizedPlaneStrain/index.md:
```diff
@@ -0,0 +1,58 @@
+
+!listing modules/solid_mechanics/test/tests/generalized_plane_strain/generalized_plane_strain_squares.i block=Physics/SolidMechanics/GeneralizedPlaneStrain
+
+An example of using generalized plane strain action through the Solid Mechanics quasi-static physics with a different `out_of_plane_direction` than the default of $z$  is given by:
```
done

bwspenc replied:
I don't see this changed.

GiudGiud commented on modules/solid_mechanics/doc/content/syntax/Physics/SolidMechanics/QuasiStatic/index.md:
```diff
@@ -0,0 +1,106 @@
+
+!syntax description /Physics/SolidMechanics/QuasiStatic/SolidMechanicsAction
+
+The Solid Mechanics Physics is a convenience object that simplifies part of the
```
done

bwspenc replied:
I don't see this changed.

GiudGiud commented on modules/solid_mechanics/doc/content/syntax/Physics/SolidMechanics/QuasiStatic/index.md:
```diff
@@ -0,0 +1,106 @@
+
+## Constructed MooseObjects
+
+The Solid Mechanics QuasiStatic Physics is used to construct the kernels, displacement variables, and strain materials in a consistent manner as required for a continuum mechanics simulation simulation. Optionally it generates aux variables and auxkernels to aid in the output of tensor components and scalar quantities.
```
done

bwspenc replied:
I don't see this changed

GiudGiud commented on modules/solid_mechanics/doc/content/syntax/Physics/SolidMechanics/QuasiStatic/index.md:
```diff
@@ -0,0 +1,106 @@
+
+### For the New Lagrangian Kernel system
+
+!table id=tmMaster_action_table_lc caption=Correspondence Among Action Functionality and MooseObjects for the Solid Mechanics QuasiStatic Physics, new kernel system
```
done

GiudGiud commented on modules/solid_mechanics/doc/content/syntax/Physics/SolidMechanics/QuasiStatic/index.md:
```diff
@@ -0,0 +1,106 @@
+
+### Subblocks
+
+The subblocks of the QuasiStatic Physics are what triggers MOOSE objects to be built.
```
done
yeah it s not a thing yet. Once we have more physics and SM is a physics too, I think we can go back from "Physics Action" to "Physics"

GiudGiud commented on framework/doc/content/automatic_differentiation/index.md:
```diff
@@ -297,7 +297,7 @@ interface. Material properties with automatic differentiation info can be
 properties can be retrieved in compute objects like `ADKernels` by using the
 `getADMaterialProperty` API. For detailed examples of [!ac](AD) use, the reader is
-encouraged to investigate the tensor mechanics, navier-stokes, and level-set
+encouraged to investigate the solid mechanics, navier-stokes, and level-set
```
no I think it looks more English not to hyphenate

GiudGiud commented on framework/doc/content/syntax/NodalKernels/index.md:
```diff
@@ -12,7 +12,7 @@ Some examples include:
   - [UpperBoundNodalKernel.md]
 
-- assigning mass to nodes instead of elements, as performed in the Tensor Mechanics module
+- assigning mass to nodes instead of elements, as performed in the Solid Mechanics module
```
Changed the framework doc but the module doc is widely using 'Solid Mechanics' module (from 'Tensor Mechanics' module)

GiudGiud commented on modules/doc/content/newsletter/2022/2022_02.md:
```diff
@@ -68,7 +68,7 @@ simulations with very large rotations or stretches.
       caption=Large stretch and rotations simulation using the Lagrangian kernels.
 
-The Lagrangian kernel system is included in the updated [solid mechanics action](TensorMechanicsAction.md)
+The Lagrangian kernel system is included in the updated [solid mechanics action](Modules/TensorMechanics/index.md)
```
ok I ll review all newsletter changes

GiudGiud replied:
done

GiudGiud commented on modules/doc/content/newsletter/2018/2018_06.md:
```diff
@@ -57,7 +57,7 @@ A new global strain [calculation](/GlobalStrainAction.md) approach has been impl
 ## TensorMechanics Documentation
 
-The [Tensor Mechanics](modules/tensor_mechanics/index.md) module has a more complete
+The [Tensor Mechanics](modules/solid_mechanics/index.md) module has a more complete
```
the text is still Tensor Mechanics. I m not sure we want to remove links when they are still taking you to the right place?

GiudGiud replied:
done

GiudGiud commented on modules/solid_mechanics/doc/content/source/actions/TensorMechanicsAction.md:
```diff
@@ -2,7 +2,7 @@
 ## Description
 
-The solid mechanics Master action simplifies the input file syntax for creating a solid mechanics model. See the description, example use, and parameters on the [TensorMechanics/Master](/Physics/TensorMechanics/Master/index.md) system page.
+The solid mechanics Master action simplifies the input file syntax for creating a solid mechanics model. See the description, example use, and parameters on the [TensorMechanics/Master](/Modules/TensorMechanics/Master/index.md) system page.
```
done. This is previous doco not mine

bwspenc replied:
This didn't get fixed. I realize that you didn't touch that, but I think it's a worthwhile change.

GiudGiud commented on modules/fsi/doc/content/modules/fsi/sqa/fsi_srs.md:
```diff
@@ -22,7 +22,7 @@ mechanics module may be used for the solid domain.
 For more context on this point, the {{module}} module SRS defers to the framework
 [framework_srs.md#assumptions-and-dependencies] and [Navier-Stokes](navier_stokes_srs.md#assumptions-and-dependencies)
-and [tensor mechanics](tensor_mechanics_srs.md#assumptions-and-dependencies) module assumptions and
+and [solid mechanics](solid_mechanics_srs.md#assumptions-and-dependencies) module assumptions and
```
same, cannot

bwspenc replied:
The point of all these comments I made was that it's just going to confuse people if you go back and rewrite history. I would lean on the side of keeping the text as it was and getting rid of the hyperlinks, because I don't think they're all that important. I don't really care that much, though, so we can just disagree.

GiudGiud commented on modules/doc/content/newsletter/2024/2024_01.md:
```diff
@@ -114,4 +114,4 @@ moose-wasp               2023.11.29        build_0
 - Extra element IDs can now be used to assign elements to subdomains in the [ParsedSubdomainMeshGenerator.md].
 - The [PropertyReadFile.md] user object can now read multiple files, once every time it is executed. The execution is controlled with the [!param](/UserObjects/PropertyReadFile/execute_on) parameter.
-- The [NEML2](syntax/NEML2/index.md) constitutive model library was added to the tensor mechanics module as an optional dependency. NEML2 is built on top of libTorch and supports the implementation of device-independent material models (with GPU support).
+- The [NEML2](syntax/NEML2/index.md) constitutive model library was added to the solid mechanics module as an optional dependency. NEML2 is built on top of libTorch and supports the implementation of device-independent material models (with GPU support).
```
NEML2 was merged one month ago, I dont see the value in referring to the old module here

GiudGiud commented on modules/doc/content/newsletter/2022/2022_02.md:
```diff
@@ -34,14 +34,14 @@ key features of this module include:
 
 A new system of solid mechanics kernels are available in the
-[tensor mechanics](modules/tensor_mechanics/index.md) module.
+[solid mechanics](modules/solid_mechanics/index.md) module.
```
done

GiudGiud commented on modules/doc/content/newsletter/2020/2020_01.md:
```diff
@@ -72,7 +72,7 @@ with the two domains interacting through contact constraints.
 ## Shell Element for Mechanics
 
-A four-noded [shell element](modules/tensor_mechanics/ShellElements.md) has been added to
+A four-noded [shell element](modules/solid_mechanics/ShellElements.md) has been added to
```
it s still the right page

GiudGiud commented on modules/doc/content/newsletter/2019/2019_12.md:
```diff
@@ -23,10 +23,10 @@ To upgrade to use Python 3, please update your MOOSE environment package by foll
 ## Solid Mechanics module retirement schedule
 
-The solid mechanics capabilities in the MOOSE modules were originally provided by the `solid_mechanics` module. Because of limitations with the architecture of the material models in that module, a decision was made several years to re-implement the mechanics capabilities in a new module called `tensor_mechanics`. After a significant development effort, we have migrated all of the models in the I ...
+The solid mechanics capabilities in the MOOSE modules were originally provided by the `solid_mechanics` module. Because of limitations with the architecture of the material models in that module, a decision was made several years to re-implement the mechanics capabilities in a new module called `solid_mechanics`. After a significant development effort, we have migrated all of the models in the IN ...
```
done

GiudGiud commented on modules/doc/content/newsletter/2018/2018_05.md:
```diff
@@ -13,7 +13,7 @@ see [/ActuallyExplicitEuler.md]
 MOOSE [now supports](https://github.com/idaholab/moose/pull/10238) vector finite elements. Currently
 available element types are `LAGRANGE_VEC` and `NEDELEC_ONE`. The former is equivalent to a vector
-of scalar `LAGRANGE` elements and is useful in applications like tensor mechanics or Navier-Stokes
+of scalar `LAGRANGE` elements and is useful in applications like solid mechanics or Navier-Stokes
```
done

GiudGiud committed:
Address Ben's review
- revert some module name changes in newsletters
- improve reference spelling to the solid mechanics module

part 2

GiudGiud committed:
Allow actions registered to meta-action tasks to create an action
that has a valid and a deprecated syntax and still have the single-syntax-action behavior

bwspenc commented on modules/solid_mechanics/doc/content/syntax/Physics/SolidMechanics/QuasiStatic/index.md:
```diff
@@ -0,0 +1,106 @@
+
+### For the `StressDivergenceTensors` Kernels
+
+!table id=tmMaster_action_table_sdt caption=Correspondence Among Action Functionality and MooseObjects for the Solid Mechanics QuasiStatic Physics, current kernel system
```
I don't see this changed

bwspenc reviewed (approved):
All the issues I raised have been addressed. There are still some failing tests, so there are some minor things that would need to be fixed before merging. I'm happy to re-review when it's ready.

GiudGiud committed:
Update coverage target to new module name

GiudGiud committed:
Update some UEL and NEML2 paths
And a few tests doco paths

GiudGiud committed:
Change documentation test makefile to exclude tensor mechanics symlink folder

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2082660/) on fcffe59 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>d6b47a</th>
      <th colspan=4>#26739 fcffe5</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.29%</td>
      <td>85.29%</td>
      <td>+0.00%</td>
      <td>77.78%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>100795</td>
      <td>100802</td>
      <td>+7</td>
      <td>7</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>17385</td>
      <td>17386</td>
      <td>+1</td>
      <td>2</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage/framework)


## Modules coverage

Inconsistent report tags were found between the head and base reports.
This can happen when reports are missing from either the head or the base.

Inconsistent tags:
  module-solid_mechanics
  module-solid_mechanics-parallel

### Combined

<table>
  <thead>
    <tr>
      <th></th>
      <th>d6b47a</th>
      <th colspan=4>#26739 fcffe5</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>48.81%</td>
      <td>48.81%</td>
      <td>-</td>
      <td>50.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>41</td>
      <td>41</td>
      <td>-</td>
      <td>1</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>43</td>
      <td>43</td>
      <td>-</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage_diff/combined)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage/combined)

### Contact

<table>
  <thead>
    <tr>
      <th></th>
      <th>d6b47a</th>
      <th colspan=4>#26739 fcffe5</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>90.31%</td>
      <td>90.31%</td>
      <td>-</td>
      <td>25.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>4456</td>
      <td>4456</td>
      <td>-</td>
      <td>1</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>478</td>
      <td>478</td>
      <td>-</td>
      <td>3</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage_diff/contact)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage/contact)

### Fsi

<table>
  <thead>
    <tr>
      <th></th>
      <th>d6b47a</th>
      <th colspan=4>#26739 fcffe5</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>87.43%</td>
      <td>87.43%</td>
      <td>-</td>
      <td>50.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>327</td>
      <td>327</td>
      <td>-</td>
      <td>1</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>47</td>
      <td>47</td>
      <td>-</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage_diff/fsi)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage/fsi)

### Peridynamics

<table>
  <thead>
    <tr>
      <th></th>
      <th>d6b47a</th>
      <th colspan=4>#26739 fcffe5</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>77.46%</td>
      <td>77.46%</td>
      <td>-</td>
      <td>25.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>3068</td>
      <td>3068</td>
      <td>-</td>
      <td>1</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>893</td>
      <td>893</td>
      <td>-</td>
      <td>3</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage_diff/peridynamics)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage/peridynamics)

### Porous flow

<table>
  <thead>
    <tr>
      <th></th>
      <th>d6b47a</th>
      <th colspan=4>#26739 fcffe5</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>95.35%</td>
      <td>95.35%</td>
      <td>-</td>
      <td>25.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>11251</td>
      <td>11251</td>
      <td>-</td>
      <td>1</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>549</td>
      <td>549</td>
      <td>-</td>
      <td>3</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage_diff/porous_flow)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage/porous_flow)

### Xfem

<table>
  <thead>
    <tr>
      <th></th>
      <th>d6b47a</th>
      <th colspan=4>#26739 fcffe5</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.27%</td>
      <td>82.27%</td>
      <td>-</td>
      <td>25.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>6836</td>
      <td>6836</td>
      <td>-</td>
      <td>1</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1473</td>
      <td>1473</td>
      <td>-</td>
      <td>3</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage_diff/xfem)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26739/coverage/xfem)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/26739/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 77.78% is less than the suggested 90.0%
- `combined` new line coverage rate 50.00% is less than the suggested 90.0%
- `contact` new line coverage rate 25.00% is less than the suggested 90.0%
- `fsi` new line coverage rate 50.00% is less than the suggested 90.0%
- `peridynamics` new line coverage rate 25.00% is less than the suggested 90.0%
- `porous_flow` new line coverage rate 25.00% is less than the suggested 90.0%
- `xfem` new line coverage rate 25.00% is less than the suggested 90.0%



This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2082660/) on fcffe59 wanted to post the following:


The following coverage requirement(s) failed:
- `solid_mechanics` coverage rate 82.75% is less than the required 84.0%

GiudGiud commented:
Rebased on devel to get an updated coverage number as the diff failed. 
It would be good to know why the 2% drop. If it can be explained (and it s ok), the coverage number is updated in the last commit

GiudGiud commented:
Coverage issue resolved

This is as many tests we can pass without updating the apps downstream. 

Merging
